code/obj/machinery/computer/cloning.dm

```diff
@@ -77,20 +78,23 @@
 	..()
 	START_TRACKING
 	SPAWN_DBG(0.7 SECONDS)
-		if(portable) return
-		src.scanner = locate(/obj/machinery/clone_scanner, orange(2,src))
-		src.pod1 = locate(/obj/machinery/clonepod, orange(4,src))
-
-		var/hookup_error = FALSE
-		if (isnull(src.scanner))
-			hookup_error = TRUE
-		if (isnull(src.pod1))
-			hookup_error = TRUE
-		if (!hookup_error)
-			src.pod1?.connected = src
-			src.scanner?.connected = src
+		connection_scan()
 	return
 
+/obj/machinery/computer/cloning/connection_scan()
```

Comment:
If current `scanner` and `pod1` exist you should first clean up their `connected` vars here. (Otherwise you might a scanner or a pod thinking that it's connected to a computer but the computer disagreeing.)